docs/source/java/substrait.rst

```diff
@@ -148,297 +136,60 @@ This Java program:
     import org.apache.arrow.memory.BufferAllocator;
     import org.apache.arrow.memory.RootAllocator;
     import org.apache.arrow.vector.ipc.ArrowReader;
+    import org.apache.calcite.sql.parser.SqlParseException;
+
+    import java.nio.ByteBuffer;
+    import java.util.Base64;
+    import java.util.Optional;
 
     public class ClientSubstraitExtendedExpressionsCookbook {
 
-      public static void main(String[] args) throws Exception {
-        // project and filter dataset using extended expression definition - 03 Expressions:
-        // Expression 01 - CONCAT: N_NAME || ' - ' || N_COMMENT = col 1 || ' - ' || col 3
-        // Expression 02 - ADD: N_REGIONKEY + 10 = col 1 + 10
-        // Expression 03 - FILTER: N_NATIONKEY > 18 = col 3 > 18
+      public static void main(String[] args) throws SqlParseException {
         projectAndFilterDataset();
       }
 
-      public static void projectAndFilterDataset() {
+      private static void projectAndFilterDataset() throws SqlParseException {
         String uri = "file:///Users/data/tpch_parquet/nation.parquet";
-        ScanOptions options = new ScanOptions.Builder(/*batchSize*/ 32768)
-            .columns(Optional.empty())
-            .substraitFilter(getSubstraitExpressionFilter())
-            .substraitProjection(getSubstraitExpressionProjection())
-            .build();
-        try (
-            BufferAllocator allocator = new RootAllocator();
-            DatasetFactory datasetFactory = new FileSystemDatasetFactory(
-                allocator, NativeMemoryPool.getDefault(),
-                FileFormat.PARQUET, uri);
-            Dataset dataset = datasetFactory.finish();
-            Scanner scanner = dataset.newScan(options);
-            ArrowReader reader = scanner.scanBatches()
-        ) {
+        ScanOptions options =
+            new ScanOptions.Builder(/*batchSize*/ 32768)
+                .columns(Optional.empty())
+                .substraitFilter(getByteBuffer(new String[]{"N_NATIONKEY > 18"}))
+                .substraitProjection(getByteBuffer(new String[]{"N_REGIONKEY + 10",
+                    "N_NAME || CAST(' - ' as VARCHAR) || N_COMMENT"}))
+                .build();
+        try (BufferAllocator allocator = new RootAllocator();
+             DatasetFactory datasetFactory =
+                 new FileSystemDatasetFactory(
+                     allocator, NativeMemoryPool.getDefault(), FileFormat.PARQUET, uri);
+             Dataset dataset = datasetFactory.finish();
+             Scanner scanner = dataset.newScan(options);
+             ArrowReader reader = scanner.scanBatches()) {
           while (reader.loadNextBatch()) {
-            System.out.println(
-                reader.getVectorSchemaRoot().contentToTSVString());
+            System.out.println(reader.getVectorSchemaRoot().contentToTSVString());
```

Comment:
nit: does this need formatting. I am merely asking this comparing with the existing code.

Reply:
I'll review how a linter will be configured/implemented for Java code.